File: app/src/main/java/dev/radley/omgstarwars/model/sw/Vehicle.java
package dev.radley.omgstarwars.model.sw;


import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.ArrayList;

import dev.radley.omgstarwars.R;

/**
 * Created by Oleur on 22/12/2014.
 * Vehicle model represents a single transport craft that does not have hyperdrive capability.
 */
public class Vehicle extends SWModel implements Serializable {

    public String model;
    public String manufacturer;
    public String length;
    public String crew;
    public String passengers;
    public String consumables;

    @SerializedName("vehicle_class")
    public String vehicleClass;

    @SerializedName("cost_in_credits")
    public String costInCredits;

    @SerializedName("max_atmosphering_speed")
    public String maxAtmospheringSpeed;

    @SerializedName("cargo_capacity")
    public String cargoCapacity;

    @SerializedName("pilots")
    public ArrayList<String> pilotsUrls;

    @SerializedName("films")
    public ArrayList<String> filmsUrls;
}

File: OMGStarWars/app/src/main/java/dev/radley/omgstarwars/models/Vehicle.kt
package dev.radley.omgstarwars.models


import com.google.gson.annotations.SerializedName

import java.io.Serializable
import java.util.ArrayList

/**
 * Vehicle model represents a single transport craft that does not have hyperdrive capability.
 */
open class Vehicle : SWModel(), Serializable {

    var model: String = ""
    var manufacturer: String = ""
    var length: String = ""
    var crew: String = ""
    var passengers: String = ""
    var consumables: String = ""

    override val subtitle: String
        get() = model

    @SerializedName("vehicle_class")
    var vehicleClass: String = ""

    @SerializedName("cost_in_credits")
    var costInCredits: String = ""

    @SerializedName("max_atmosphering_speed")
    var maxAtmospheringSpeed: String = ""

    @SerializedName("cargo_capacity")
    var cargoCapacity: String = ""

    @SerializedName("pilots")
    override var relatedPeople: ArrayList<String>? = null

    @SerializedName("films")
    override var relatedFilms: ArrayList<String>? = null

    override val relatedPeopleTitle: String
        get() = "Pilots"
}